makefiles/auto_init.inc.mk

```diff
@@ -0,0 +1,16 @@
+# Determine the list of modules that provide auto_init from the list of all
```

Comment:
I think this could be generalized (no need to use AUTO_INIT...). It is actually very similar to the submodule logic, which also has a configurable prefix, bit wouldn't work out of the box.

But that's for $future. :)